nansat/mappers/mapper_sentinel1_l1.py

```diff
@@ -29,189 +29,145 @@
             raise WrongMapperError(filename)
 
-        mdsDict = {}
-        for ff in mdsFiles:
-            mdsDict[
-                os.path.splitext(os.path.basename(ff))[0].split('-')[3]] = ff
-
-        self.calXMLDict = {}
-        for ff in calFiles:
-            self.calXMLDict[
-                os.path.splitext(
-                os.path.basename(ff))[0].split('-')[4]] = self.read_vsi(ff)
-
-        self.noiseXMLDict = {}
-        for ff in noiseFiles:
-            self.noiseXMLDict[
-                os.path.splitext(
-                os.path.basename(ff))[0].split('-')[4]] = self.read_vsi(ff)
-
-        self.annotationXMLDict = {}
-        for ff in annotationFiles:
-            self.annotationXMLDict[
-                os.path.splitext(
-                os.path.basename(ff))[0].split('-')[3]] = self.read_vsi(ff)
-
-        self.manifestXML = self.read_vsi(manifestFile[0])
-
-        missionName = {'S1A': 'SENTINEL-1A', 'S1B': 'SENTINEL-1B'}[
-            os.path.split(filename.rstrip('/'))[1][:3]]
-
-        # very fast constructor without any bands
-        if manifestonly:
-            self.init_from_manifest_only(self.manifestXML,
-                                         self.annotationXMLDict[
-                                         list(self.annotationXMLDict.keys())[0]],
-                                         missionName)
-            return
+        # convert list of MDS files into dictionary. Keys - polarizations in upper case.
+        mds_files = {os.path.basename(ff).split('-')[3].upper():ff for ff in mds_files}
+        polarizations = list(mds_files.keys())
 
-        gdalDatasets = {}
-        for key in mdsDict.keys():
-            # Open data files
-            gdalDatasets[key] = gdal.Open(mdsDict[key])
+        # read annotation files
+        annotation_data = self.read_annotation(annotation_files)
+        if not fast:
+            annotation_data = Mapper.correct_geolocation_data(annotation_data)
 
-        if not gdalDatasets:
-            raise WrongMapperError('%s: No Sentinel-1 datasets found'
-                    %mdsDict[key])
+        # read manifest file
+        manifest_data = self.read_manifest_data(manifest_files[0])
 
-        # Check metadata to confirm it is Sentinel-1 L1
-        metadata = gdalDatasets[list(mdsDict.keys())[0]].GetMetadata()
+        # very fast constructor without any bands only with some metadata and geolocation
+        self._init_empty(manifest_data, annotation_data)
 
-        if not 'TIFFTAG_IMAGEDESCRIPTION' in metadata.keys():
-            raise WrongMapperError(filename)
-        if (not 'Sentinel-1' in metadata['TIFFTAG_IMAGEDESCRIPTION']
-                and not 'L1' in metadata['TIFFTAG_IMAGEDESCRIPTION']):
-            raise WrongMapperError(filename)
+        # skip adding bands in the fast mode and RETURN
+        if fast:
+            return
 
         warnings.warn('Sentinel-1 level-1 mapper is not yet adapted to '
```

Comment:
is this warning needed still?

Reply:
I don't know. You added it. Shall it be removed?

Reply:
not sure.... I guess it doesn't work for complex data, does it? Perhaps it can be removed anyway. It is a bit annoying...